### ISSUE TITLE
Possible bug in dmd for Windows converting known numbers into -nan

### ISSUE THREAD

**Issue by davidomarf** [daque-dev/viare#3]
This problems was spotted first when running `dub run` after pulling https://github.com/daque-dev/viare/commit/36e47004629bbe8900da9570c0271c5a18badc57.

First, [`source/app.d`](https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/app.d) calls `heightmapPerspectiveTest()` defined at [`source/tests/heightmap-perspective.d`](https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/tests/heightmap-perspective.d#L36)

heightmapPerspectiveTest(), at [line 68](https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/tests/heightmap-perspective.d#L68), calls `fillByHeightFunction()`, defined as

https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/heightmap/heightmap.d#L40-L45

using `QuevHeightFunction()` as argument for `heightFunction`, defined in [`source/heightmap/quev.d`](https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/heightmap/quev.d#L80-L100).

`QuevHeightFunction()`  is specified to receive `(double x, double y)` as arguments. 

https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/heightmap/quev.d#L102

And fillByHeightFunction was sending them. 

https://github.com/daque-dev/viare/blob/36e47004629bbe8900da9570c0271c5a18badc57/source/heightmap/heightmap.d#L44

At least theoretically.

---

At the first iteration, `QuevHeightFunction()` should be receiving `0, 0`, but it was receiving `-nan, 0`.

My conclusion was that error must have been coming from `fillByHeightFunction()`.

Before calling `heightFunction(cast(double) i / m_xlength, cast(double) j / m_ylength);`, all `i`, `m_xlength`, `j`, `m_ylength` were numbers.  

**And so were the operations `i / m_xlength` and ` j / m_ylength`.**

**And so were the casts `cast(double) i / m_xlength` and `cast(double) j / m_ylength)`.**

---

**tl;dr**

`QuevHeightFunction()` is called with two floats `(x, y)`. But it receives `(-nan, y)`.

This suggests that something happens at the moment of the function call.

## Possible Solution
The only solution that I've found so far was writing the value of `cast(double) i / m_xlength` to console before sending it to `QuevHeightFunction()`.

Yes. Only that. **Printing.**

## Steps to Reproduce

(Assuming you've already cloned the repository)

1. `git pull origin master`
2. `git checkout heightmap-perspective-test`
3. `git checkout 36e47004629bbe8900da9570c0271c5a18badc57`
4. `dub run`

## Environment
* DMD: `v2.081.1`
* DUB: `1.10.0`
* OS: Windows 10 1803

**Comment by davidomarf** [comment]
Implied in the title, but need to remark:

**Apparently, this only happens on Windows. Same commit in Linux runs ok.**

**Comment by davidomarf** [comment]
I will close the issue now, as it appears to be solved ussing a dirty fix — https://github.com/daque-dev/viare/commit/b1ad7d27c69e4e7ade7e406d9933b0c906422bda —.

Keep in mind this is not optimal solution, and it must be addressed later.
